### PR TITLE
Allow job files to consume TUS uploads

### PR DESCRIPTION
**Pull request by jmchilton**
xref https://github.com/galaxyproject/galaxy/issues/17241

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by mira-miracoli** [comment]
Not sure if my fix for tus v2 is relevant here (just briefly had a look and saw the `@expose_api_raw_anonymous_and_sessionless` decorator)
- #17246

File: lib/galaxy/webapps/galaxy/api/job_files.py [hunk 1]
```diff
@@ -99,6 +100,14 @@ def create(self, trans, job_id, payload, **kwargs):
                 upload_store
             ), f"Filename provided by nginx ({file_path}) is not in correct directory ({upload_store})"
             input_file = open(file_path)
+        elif "session_id" in payload:
+            # code stolen from basic.py
+            session_id = payload["session_id"]
+            upload_store = trans.app.config.tus_upload_store or trans.app.config.new_file_path
```
Comment (mvdbeek) [inline lib/galaxy/webapps/galaxy/api/job_files.py]: This should probably not be mixed into tus_upload_store ? Add a new config option for `tus_job_files_store` or add a subdir `/job_files` ?

File: lib/galaxy/webapps/galaxy/api/job_files.py [hunk 2]
```diff
@@ -114,6 +123,47 @@ def create(self, trans, job_id, payload, **kwargs):
                 pass
         return {"message": "ok"}
 
+    @expose_api_raw_anonymous_and_sessionless
```
Comment (mvdbeek) [inline lib/galaxy/webapps/galaxy/api/job_files.py]: I think Mira is right here, we probably need to return valid JSON for TUS 2.0, so `@expose_api_anonymous_and_sessionless` is the right decorator here and for `tus_hooks`

**Review by mvdbeek — APPROVED** [review]

File: lib/galaxy/webapps/galaxy/api/job_files.py (hunk 1 above)
Comment (jmchilton) [reply lib/galaxy/webapps/galaxy/api/job_files.py]: I was somewhat certain people would say I was over engineering if I did that.

**Comment by mvdbeek** [comment]
That's awesome, thanks a lot @jmchilton!

**Pull request merged**